zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala

```diff
@@ -198,11 +198,18 @@ final case class Endpoint[PathInput, Input, Err, Output, Auth <: AuthType](
   def auth[Auth0 <: AuthType](auth: Auth0): Endpoint[PathInput, Input, Err, Output, Auth0] =
     copy(authType = auth)
 
-  def authScopes: List[String] = authType match {
-    case AuthType.ScopedAuth(_, scopes) => scopes
-    case _                              => Nil
+  def authScopes(authType: AuthType): List[String] = authType match {
```

Comment:
The recursive call should only be happening in a private method. The public one should have no params. Also to keep the duality to similar methods, it should be named `scopes`